discovery.go

```diff
@@ -0,0 +1,45 @@
+package oidc
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"strings"
+)
+
+const (
+	DiscoveryEndpoint = "/.well-known/openid-configuration"
+)
+
+type DiscoveryConfiguration struct {
+	Issuer                      string `json:"issuer,omitempty"`
+	AuthorizationEndpoint       string `json:"authorization_endpoint,omitempty"`
+	TokenEndpoint               string `json:"token_endpoint,omitempty"`
+	IntrospectionEndpoint       string `json:"introspection_endpoint,omitempty"`
+	UserinfoEndpoint            string `json:"userinfo_endpoint,omitempty"`
+	RevocationEndpoint          string `json:"revocation_endpoint,omitempty"`
+	DeviceAuthorizationEndpoint string `json:"device_authorization_endpoint,omitempty"`
+	JwksURI                     string `json:"jwks_uri,omitempty"`
+}
+
+func discover(ctx context.Context, issuer string, httpClient *http.Client, wellKnownUrl ...string) (*DiscoveryConfiguration, error) {
```

Comment:
Why do you accept a slice of strings for wellKnownUrl and not just a single string?

Reply:
This allows it to be an optional parameter, it is a fairly common Go idiom to achieve optionality, in this case a variadic function. There are others for other situations

https://bojanz.github.io/optional-parameters-go/